### DISCUSSION TITLE
Plugins not available without OpenAI API key & Stable Diffusion not working as expected

### DISCUSSION THREAD

**Discussion by Kryszn0** [danny-avila/LibreChat#2625] [Troubleshooting]
### What happened?

I am trying to setup entirely locally hosted AI chat:
- Ollama as LLM
- LiteLLM as Proxy for OpenAI
- LibreChat as interface
- Stable Diffusion for image generation

All the above is working. However, having issues when trying to incorporate Stable Diffusion into the mix.

### Steps to Reproduce

Problem 1: Plugins not available without OpenAI API key

When this variable is not set, the Plugins are not available. I'm only able to download the Stable Diffusion plugin once I add a random API key.

If I set the API key I get both drop down options for OpenAI and Plugins. NOTE: Texting anything to the OpenAI drop down results in errors/does not work. 

Problem 2: Stable Diffusion plugin not working as expected
When I go to the plugin to ask it a question, the LLM responds and it appears to try and generate an image but nothing is actually provided. I get the following:

![image](https://github.com/danny-avila/LibreChat/assets/113187314/2ad34a05-6edc-49c3-b7b7-139b36b60ab6)

Is there a way to use plugins without OpenAI? For example: Have Ollama/LiteLLM be set as the default LLM that then calls the Stable Diffusion URL when necessary?

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
You are bound to receive errors because your setup is not officially tested. I'll be working on tool usage for non-openai endpoints soon

**Reply by Kryszn0** [reply]
Ahh gotchya

Looking forward to that update! Hoping LibreChat can become my all-in-one UI for a full-featured self-hosted AI

**Reply by ghost** [reply]
hey @danny-avila is there something litellm-related causing this? Happy to help

**Reply by wallartup** [reply]
Great stuff with Ollama based plugins.

**Comment by oatmealm** [comment]
I’ve installed librechat yesterday and was hoping to set it up local first as well. Im using ollama for most of my work with other frontends. @K-J-VV  would love to see your configuration so far as it seems it’ll require quite a lot of fiddling to remove the default commercial endpoints.
